File: back-end/src/main/java/pl/straburzynski/campfiresongs/song/model/Song.java
package pl.straburzynski.campfiresongs.song.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.UUID;

@Entity
@Table(name = "SONG")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Song {

    @Id
    @Column(name = "ID")
    @Type(type = "uuid-char")
    private UUID id;

    @Column(name = "AUTHOR")
    private String author;

    @Column(name = "TITLE")
    private String title;

    @Column(name = "LYRICS", length = Integer.MAX_VALUE)
    private String lyrics;

}

File: back-end-kotlin/src/main/kotlin/pl/straburzynski/campfiresongs/song/model/Song.kt
package pl.straburzynski.campfiresongs.song.model


import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.UuidGenerator
import java.util.UUID

@Entity
@Table(name = "SONGS")
data class Song(
    @Id
    @Column(name = "ID", nullable = false)
    @UuidGenerator
    var id: UUID? = null,

    @Column(name = "AUTHOR")
    val author: String,

    @Column(name = "TITLE")
    val title: String,

    @Column(name = "LYRICS", length = Int.MAX_VALUE)
    val lyrics: String
)
